src/coreclr/src/System.Private.CoreLib/ILLinker/ILLink.Suppressions.cs

```diff
@@ -0,0 +1,276 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: UnconditionalSuppressMessage ("ILLinker", "IL2006", Scope = "member", Target = "F:System.LazyDebugView`1._lazy")]
+[assembly: UnconditionalSuppressMessage ("ILLinker", "IL2006", Scope = "member", Target = "M:Internal.Runtime.InteropServices.ComActivator.BasicClassFactory.CreateInstance(System.Object,System.Guid@,System.Object@)")]
+[assembly: UnconditionalSuppressMessage ("ILLinker", "IL2006", Scope = "member", Target = "M:Internal.Runtime.InteropServices.ComActivator.ClassRegistrationScenarioForType(Internal.Runtime.InteropServices.ComActivationContext,System.Boolean)")]
+[assembly: UnconditionalSuppressMessage ("ILLinker", "IL2006", Scope = "member", Target = "M:Internal.Runtime.InteropServices.ComponentActivator.InternalGetFunctionPointer(System.Runtime.Loader.AssemblyLoadContext,System.String,System.String,System.IntPtr)")]
+[assembly: UnconditionalSuppressMessage ("ILLinker", "IL2006", Scope = "member", Target = "M:Internal.Runtime.InteropServices.LicenseInteropProxy.#ctor")]
```

Comment:
How are we running linker on this (what's the command line)? Linker should be able to analyze this. I've seen this warning show up when linker isn't passed a reference to System.ComponentMode.TypeConverter assembly and can't analyze the reflection as a result.

(There's multiple instances of the same problem within this suppression file.)

Reply:
I think these are the arguments that are passed when linking CoreLib:

```
-reference "D:\source\repos\runtime\artifacts\obj\coreclr\System.Private.CoreLib\x64\Debug\PreTrim/System.Private.CoreLib.dll"
-out "D:\source\repos\runtime\artifacts\obj\coreclr\System.Private.CoreLib\x64\Debug"
-r System.Private.CoreLib -c skip -u skip -p link System.Private.CoreLib -t -b true -v true --strip-descriptors false --strip-substitutions false --skip-unresolved true --disable-opt unusedinterfaces --keep-dep-attributes true
```

Reply:
Is there any chance we can have linker see the whole framework when trimming the individual libraries? (I think this comes back to an old topic...)

Without that, we're going to see warnings for all `Type.GetType("Foo, X").GetMethod("Bar")` patterns because even though linker could know what Foo refers to, it won't be able to resolve that and has to treat it as "don't know". We'll never know if that reflection is actually safe (linker will be able to reason about it for actual app build) or not (linker can no longer reason about it when seeing the whole framework either because e.g. someone refactored things).

Reply:
> Is there any chance we can have linker see the whole framework when trimming the individual libraries?

This approach was rejected in https://github.com/dotnet/runtime/issues/31712

Reply:
Do we need to revisit that? What are the options to avoid the bogus warnings like this one?

Reply:
> Do we need to revisit that?

In my mind, this is no different than the advantages already outlined in that issue - https://github.com/dotnet/runtime/issues/31712#issuecomment-619270604. Specifically:

>> 1. The linker can better understand the code when it sees the entire "distributable".

This is just another case of the linker not being able to reason about everything at once.
____

> What are the options to avoid the bogus warnings like this one?

* One option is to keep doing what we are doing in this PR. It suppresses the warning.
* One option is to pass in suppress messages separately to the linker that are only for the "Library Build", just like we do with the `ILLinkTrim_LibraryBuild.xml` files.
* Lastly, we could revisit #31712 and change the way we run the linker during the build to do the whole shared fx/runtimepack at once.
    - A variation on this approach is to only run the validation against the whole shared fx/runtimepack at the of the build, and throw away the outputted binaries. We still run the linker during the individual library build, and suppress all warnings like we currently are. The warnings only come at the end, during the full build.
        - This addresses the main concerns in #31712 that the inner dev loop produces the same binaries as the full build.

Reply:
> A variation on this approach is to only run the validation against the whole shared fx/runtimepack at the of the build, and throw away the outputted binaries

That sounds most reasonable option to me.

Reply:
Yep, if we want to do the precise analysis we have to run against the whole framework and every runtime pack. Doing it the way as it's today will rely on manual information copying which is very error-prone in our scale.

Reply:
The first two options (suppress warnings unconditionally or suppress warning during library build) run the risk that if something happens and linker stops recognizing the pattern we either don't see the warning at all, or only _customers_ see the warnings. That's quite undesirable for things as fragile as "app-specific customization of the framework" (which is what trimming+feature switches really is - no two frameworks will be the same and it's hard to prove correctness without automation help).

I like option 3 the best too even though it's probably more work.

Reply:
As much as I don't like running the same tool twice, it sounds as the best option.